app/controllers/application_controller.rb

```diff
@@ -7,4 +11,14 @@ def render_json_api_errors(object)
            adapter:    :json_api,
            serializer: ActiveModel::Serializer::ErrorSerializer
   end
+
+  def authenticate_user_from_token!
+    authenticate_with_http_token do |token, options|
+      user_email = options[:email].presence
+      user       = user_email && User.find_by_email(user_email)
+      if user && Devise.secure_compare(user.authentication_token, token)
```

Comment:
we should rotate the frontend user tokens from time to time... I am thinking in wiping the token column once the user signs off, or after an amount of time since the last connection